magellan-library/src/main/java/com/wealthfront/magellan/navigation/LazySetNavigator.kt

```diff
@@ -36,9 +39,39 @@ public open class LazySetNavigator(
   }
 
   public fun addNavigable(navigable: NavigableCompat) {
+    existingNavigables.add(navigable)
     lifecycleRegistry.attachToLifecycleWithMaxState(navigable, LifecycleLimit.CREATED)
   }
 
+  public fun removeNavigables(navigables: Set<NavigableCompat>) {
+    for (navigable in navigables) {
+      removeNavigable(navigable)
+    }
+  }
+
+  public fun removeNavigable(navigable: NavigableCompat) {
+    existingNavigables.remove(navigable)
+    lifecycleRegistry.removeFromLifecycle(navigable)
+  }
+
+  public fun safeAddNavigable(navigable: NavigableCompat) {
+    if (!existingNavigables.contains(navigable)) {
+      addNavigable(navigable)
+    }
+  }
+
+  public fun updateNavigables(navigables: Set<NavigableCompat>, handleCurrentTabRemoval: () -> Unit) {
+    val navigablesToRemove = existingNavigables subtract navigables
+    val navigablesToAdd = navigables subtract existingNavigables
+
+    if (navigablesToRemove.contains(currentNavigable)) {
+      handleCurrentTabRemoval()
+    }
```

Comment:
There's an edge case to be aware of when removing a Navigable that is currently visible. We need to switch to another navigable before we can remove it because once you call `removeFromLifecycle` on a navigable you can't call `replace` on it anymore. I wrote a test `updateMultipleNavigables_andRemoveCurrentNavigable` describing this case.